File: app/src/main/java/com/gooner10/popularmoviesapp/moviefavourite/MovieFavouriteContract.java
package com.gooner10.popularmoviesapp.moviefavourite;

import com.gooner10.popularmoviesapp.data.MovieItem;

import java.util.List;

/**
 * Contract for the favorite fragment
 */

public class MovieFavouriteContract {
    interface View {
        void displayFavoriteMovie(List<MovieItem> movieDataList);
    }

    interface Presenter {
        void loadFavoriteMovieFromRepository();
    }
}

File: app/src/main/java/com/gooner10/popularmoviesapp/moviefavourite/MovieFavouriteContract.kt
package com.gooner10.popularmoviesapp.moviefavourite

import com.gooner10.data.MovieItem

/**
 * Contract for the favorite fragment
 */

class MovieFavouriteContract {
    interface View {
        fun displayFavoriteMovie(movieDataList: List<com.gooner10.data.MovieItem>)
    }

    internal interface Presenter {
        fun loadFavoriteMovieFromRepository()
    }
}
